### PR TITLE
Create crud operations for accomodations and locations

### PR DESCRIPTION
**Pull request by muhireheir**
### What does this PR do
- Updates accomodation model
- creates these models : **rooms** **locations**
- implimesnts CRUD operations on above models
- Updates the feature of creating trip requests
### How it should be manualy tested

> `**Add the follwing in your .env file**`
> CLOUD_NAME=faustin250
> CLOUD_API_KEY=886115265422219
> CLOUD_API_SECRET=3tYIbq4FrNHFj6ix2BWqwc_uvhY

**run this command `npm run db:migrate:undo:all` and then `npm start`**

**1. LOCATIONS CRUD**
create locations11
delete locations
update locations
view locations
**2.ACCOMODATIONS CRUD** 
create accomodations
delete accomodation
update accomodation
view accomodations by location
view all accomodations

**For creating trip request , the request body should look like this **
trip request

### CODE REVIEW

File: src/controllers/accomodationsController.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      const { path } = req.files[0];
+      const image = await cloudinaryUploader(path);
+      const accomodation = {
+        name: req.body.name,
+        description: req.body.description,
+        location_id: req.body.location_id,
+        amenities: req.amenities,
+        image,
+      };
+      console.log(image);
```
Comment (nkalyesubula) [inline src/controllers/accomodationsController.js]: dont push code with console log snippets

File: src/middlewares/validators/validate.js [hunk 2]
```diff
@@ -196,11 +206,44 @@ export default class validator {
...
+    }
+  }
+
   static async checkAcommodation(req, res, next) {
     try {
-      const { accomodation } = req.body;
-      const chk = await accomodationService.findById(accomodation);
+      const { destination } = req.body;
+      const accomodation = req.body.accomodation || req.params.accomodation;
+      const chk = await accomodationService.findByIdAndLocation(accomodation, destination);
```
Comment (nkalyesubula) [inline src/middlewares/validators/validate.js]: use meaningful variable names

**Review by nkalyesubula — COMMENTED** [review]
Looks good, however use meaningful variable names

**Review by Faustin250 — APPROVED** [review]

**Review by mugisha1555 — COMMENTED** [review]
looks good

**Review by Faustin250 — APPROVED** [review]

**Review by nkalyesubula — APPROVED** [review]
LGTM

**Pull request merged**